src/Momento.Sdk/Config/Configuration.cs

```diff
@@ -7,9 +7,9 @@ namespace Momento.Sdk.Config;
 
 public class Configuration : IConfiguration
 {
-    public IRetryStrategy RetryStrategy { get; }
-    public IList<IMiddleware> Middlewares { get; }
-    public ITransportStrategy TransportStrategy { get; }
+    public IRetryStrategy RetryStrategy { get; set; }
+    public IList<IMiddleware> Middlewares { get; set; }
+    public ITransportStrategy TransportStrategy { get; set; }
```

Comment:
I'm not sure how strongly I feel about it but my inclination is to disallow setters here.  I would prefer for these config objects to be immutable, otherwise they could theoretically be changed at runtime.

See my middleware PR for an example of a `With` copy constructor that can be used to augment an existing config before it is passed to the client constructor.  Happy to discuss if you disagree.

Reply:
I'll revert to immutable and add more builder-like constructs. Let's loop back at a later date to check compatibility with the .NET configuration interfaces

Reply:
I changed this in https://github.com/momentohq/client-sdk-dotnet/pull/172/commits/d7e44627336915bb150c211df7b5d4314acd825b

Because` we depend on interfaces in the code, I had to push the `With*` builder methods into the interface. Let me know what you think.